index.bs

```diff
@@ -1742,24 +1745,23 @@ interface MLGraph {
 </dl>
 
 <dl dfn-type=method dfn-for=MLGraph>
-    : <dfn>compute(inputs, outputs)</dfn>
+    : <dfn>compute(inputs, outputNames)</dfn>
     ::
-        Issue a compute request of the {{MLGraph}} given {{MLNamedInputs}} and optional {{MLNamedOutputs}}. The returned {{Promise}} resolves when the results in {{MLNamedOutputs}} are ready to be consumed.
+        Issue a compute request of the {{MLGraph}} given {{MLNamedInputs}} and optional [=sequence=]&lt;{{DOMString}}&gt. Return {{MLNamedOutputs}}.
 
         <div algorithm=MLGraph.compute>
             **Called on:** {{MLGraph}} |this|.
 
             **Arguments:**
-            <pre class=argumentdef for="MLGraph/compute(inputs, outputs)">
+            <pre class=argumentdef for="MLGraph/compute(inputs, outputNames)">
                 |inputs|: a {{MLNamedInputs}}. The data and optional dimensions of inputs for the compute request.
-                |outputs|: an optional {{MLNamedOutputs}}. The names and pre-allocated resources of required outputs for the compute request. Default to be an empty [=record=] which means that the compute request is for all outputs.
+                |outputNames|: an optional [=sequence=]&lt;{{DOMString}}&gt. The names of required outputs for the compute request. Default to be an empty [=sequence=] which means that the compute request is for all outputs.
             </pre>
 
-            **Returns:** {{Promise}}&lt;{{MLNamedOutputs}}&gt;. The dimensions and data of outputs returned by the compute request.
+            **Returns:** {{MLNamedOutputs}}.
 
-            1. Let |promise| be [=a new promise=].
             <!-- Validate inputs and outputs -->
-            1. If any of the following requirements are unmet, then [=reject=] |promise| with a {{TypeError}} and stop.
+            1. If any of the following requirements are unmet, then throw a {{TypeError}} and stop.
```

Comment:
Would it be helpful to use different exception types to signal different ways `compute()` can fail? Do we expect a web developer to write control flow for error handling that could recover if it'd know more about the reason of failure? If that'd be a valid use case, then:

We could either use multiple simple exceptions (these match ECMAScript error objects):
https://heycam.github.io/webidl/#dfn-simple-exception

Or DOMExceptions with error names:
https://heycam.github.io/webidl/#idl-DOMException-error-names

For example, we could throw a new "DataError" DOMException to signal value.data errors.